api/v1alpha1/halayerset_types.go

```diff
@@ -62,8 +62,8 @@ type NodesSpec struct {
 
 // HALayerSetStatus defines the observed state of HALayerSet
 type HALayerSetStatus struct {
-	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	//PrevFenceAgentsSpec reflects the previous status of the Fence Agents - in order to be able to compare to the current state and decide whether an action is required
```

Comment:
IMHO this comment correctly explains what you should do, but you actually do something different:

- `in order to be able to compare to the current state` -> Yes, this is how reconcile works. Compare the current state **_of the cluster_** with the desired state, and take action.
- what you do though: compare what the current state **_should be_** by looking at an old spec

See  "What is a Level Based API" here: https://book-v1.book.kubebuilder.io/basics/what_is_a_controller.html

In case you really need to know the old spec (but IMHO should not need it): don't save it in the status. You can get the old spec in an update, just not by default. I think you can use a predicate: https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/predicate?utm_source=godoc#GenerationChangedPredicate

Reply:
After [discussion](https://coreos.slack.com/archives/CR8HZL4P3/p1641719383013200) putting this PR on hold.
Waiting for @beekhof input to decide what's the best way to go forward with it.